machAeroTutorials/aero_gridRefinementStudy.rst

```diff
@@ -66,8 +90,130 @@ An example of grid convergence plot for a family of RAE 2822 Airfoil meshes is i
+    nLayers_L2 = 80
+    s0_L2 = 4e-6
+
+    # Increasing the mesh sizes 
+    refinement=[1,2,4]
+    level =['L2','L1','L0']
+
+    for i in range(len(refinement)):
+
+        # number of points on the airfoil surface
+        nSurfPts = refinement[i]*nSurfPts_L2
+
+        # number of points on the TE.
+        nTEPts = refinement[i]*nTE_cells_L2 
+
+
+        # number of extrusion layers
+        nExtPts = refinement[i]*nLayers_L2 
+
+        # first off wall spacing
+        s0 = s0_L2/ refinement[i]
+
+        #### We can either import our desired airfoil .dat file and continue the meshing proces ####
+        #### Or we can generate the NACA airfoils if our baseline is a 4 series NACA airfoil    ####
+
+        # Read the Coordinate file
+        # filename = "n0012_old.dat"
+        # coords = readCoordFile(filename, headerlines=1)
+
+        # We can also  generate NACA 4 series airfoils
+        code='0012'
+        nPts=150
+        coords=generateNACA(code, nPts, spacingFunc=sampling.polynomial, func_args={"order": 8})
+        # print('yes',coords)
+        airfoil = coords
+
+        coords = airfoil.getSampledPts(
+        nSurfPts,
+        spacingFunc=sampling.polynomial, func_args={"order": 8},
+ 
+        nTEPts=nTEPts,
+        )
+        # print(coords)
+        # Write surface mesh
+        airfoil.writeCoords("./input/naca0012_%s" % level[i], file_format="plot3d")
+
+
+
+        options = {
+            # ---------------------------
+            #        Input Parameters
+            # ---------------------------
+            "inputFile": "./input/naca0012_%s.xyz" % level[i],
+            "unattachedEdgesAreSymmetry": False,
+            "outerFaceBC": "farfield",
+            "autoConnect": True,
+            "BC": {1: {"jLow": "zSymm", "jHigh": "zSymm"}},
+            "families": "wall",
+            # ---------------------------m
+            #        Grid Parameters
+            # ---------------------------
+            "N": nExtPts,
+            "s0": s0,
+            "marchDist": 100.0,
+
+        }
+        hyp = pyHyp(options=options)
+        hyp.run()
+        hyp.writeCGNS("./input/naca0012_%s.cgns" % level[i])
+
+
+
+As an example, the Tecplot of both cases are shown. As we can see, when we coarsen through ``cgns_utils``, the distance between each layers become higher and the growth ratio is not the same as ``prefoil`` mesh.
+
+.. figure:: images/meshexample.png
+    :scale: 40
+    :align: center
+    :alt: Mesh comparison
+    :figclass: align-center
+
+    Figure 2: Mesh comparison.
+
+.. TODO: add mesh refinement plot using this method that's similar to the RAE one
+
+Pros:
+    - It is more practical for 3D meshes since the refinement ratio is not as aggressive as ``Option 1``. This places the points on the refinement plot closer to each other  on the :math:`x`-axis so it is more likely that your coarsest volume mesh is in the asymptotic regime, which you can then use for coarse optimizations.
```

Comment:
I am a bit confused here, I thought the number of elements remained exactly the same. Maybe you meant the `y-axis`?

Also, you mention 3D meshes but here the example is with an airfoil. You should mention somewhere that you can tweak directly the surface discretization in ICEM/PointWise to replicate the same approach